### ISSUE TITLE
Adding flex col sizes seems to cause unnessary blank space to be added to the page in certain situations

### ISSUE THREAD

**Issue by Flambe** [quasarframework/quasar#16312]
### What happened?

I am trying to display q-chips in a horizontal list format inside a flex row inside a flex column. In some cases, a lot of empty space is added below the row even though there is already more than enough space to fit the chips.

![image](https://github.com/quasarframework/quasar/assets/2080851/5d07597d-745e-4ce2-86b8-89aa69d7aa6d)

### What did you expect to happen?

No extra space is added below the row when using q-chips.

### Reproduction URL

https://stackblitz.com/edit/quasarframework-9vjzes?file=src%2Fpages%2FIndexPage.vue

### How to reproduce?

1. Go to the reproduction link
2. Look at the right hand side
    * The first card shows the issue with a `.col-4` wrapping the q-chips
    * The second card shows what I would expect to happen (I've removed the `.col-4` class to replicate this)
    * The third card shows that `.col-4` works as expected when using plain spans instead of q-chips
    * The fourth card shows that is also works as expected if I remove the wrapping `.column`

This reproduction is a minimal version of what I am actually doing in my application. I've simplified it as much as I can to show the issue.

I've only tested it on browsers that I can easily access.

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar), Style & Identity (quasar)

### Platforms/Browsers

Firefox, Chrome, Microsoft Edge, Android

### Quasar info output

```Shell
Operating System - Linux(5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36) - linux/x64
NodeJs - 16.20.0

Global packages
  NPM - 9.4.2
  yarn - 1.22.19
  @quasar/cli - undefined
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.9.2 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.1.2 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.15.4 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.40 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.5
  pinia - Not installed
  vuex - Not installed
  vite - 2.9.15 -- Native-ESM powered web dev build tool
  eslint - 8.24.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - quasarframework9vjzes-ebl5
  en0 - 192.168.1.104
```

### Additional context

If this is an issue with my code and not a bug then I'll close this and make a discussion, but I can't see what could be wrong on my side.

I've tried looking into why this is happening but nothing seemed off in the CSS, so I'm not sure what could be causing this.

**Comment by pdanpdan** [comment]
This is one of the quirks of flex - the row inside the column will reserve all the space it needs to put the content on different lines.
You can fix it by setting `full-width` class on row to force it to stretch

The span example works because the spans ignore the width of their container and overflow it

**Comment by Flambe** [comment]
Ah yeah that fixed the demo! I'm still getting issues on my actual app but with that help I believe I've narrowed it down to a parent `.column` in the tree which didn't have children set up correctly. I should be able to figure it out from here I believe.

If I have any other similar issues I'll throw it in Discord/a discussion.

Thank you for the help! :) 

Edit: Managed to get mine working. Just had to check up the tree and play around with `.row` and `.full-width` to ensure everything is handled correctly
